go.mod

```diff
@@ -0,0 +1,17 @@
+module github.com/bitrise-steplib/steps-go-list
+
+go 1.16
+
+require (
```

Comment:
`go mod tidy` can be used to remove unused dependencies, it seems there are some listed here.
After that `go mod vendor` can be used to sync the vendor directory.

Reply:
added these as without them the test cases were failing ( go mod tidy does not include the modules for the tests ) - can be it's misconfiguration from my side

Reply:
This happens when we install the testing tools, installing `errcheck` adds:
- github.com/kisielk/errcheck v1.6.0 // indirect
- golang.org/x/mod v0.4.2 // indirect
- golang.org/x/sys v0.0.0-20210316092937-0b90fd5c4c48 // indirect
- golang.org/x/tools v0.1.0 // indirect

, while installing `golint` adds:
- golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5 // indirect

More details about the issue here: https://github.com/bitrise-steplib/steps-errcheck/pull/5

I think we should use our errcheck and golint steps, as they handle this issue, instead of the current scripts, so that we can remove the unused dependencies.
This is how I like to define the go-test workflow: https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/blob/master/bitrise.yml#L31-L36

Reply:
I agree that the `go-tests` wf should be replaced by what @godrei mentioned in `bitrise.yml` Can you do it @birmacher ?